manager/src/main/java/org/openremote/manager/map/MapService.java

```diff
@@ -377,6 +441,121 @@ public byte[] getMapTile(int zoom, int column, int row) {
         }
     }
 
+    public boolean saveUploadedFile(Path path, InputStream fileInputStream) {
+        Path previous = getCustomMapDataPath();
+
+        try (OutputStream outputStream = Files.newOutputStream(path)) {
```

Comment:
write output file should be done in it's own try catch so the resource is closed before you then try and open the connection on the file.